### PR TITLE
Began writing preprocessing code

### PR DESCRIPTION
**Pull request by turian**
preprocess.py will go over FSD50K and normalize the length.

transform.py is in progress, it creates random transformations on audio using SoX.

### CODE REVIEW

File: transforms.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+
+def transform_file(f):
+    transform_spec = random.choice(transforms)
+    transform = transform_spec[0]
+    params = {}
+    variable_spec = random.choice(transform_spec[1])
+    variable = variable_spec[0]
+    for param, low, hi in transform_spec[1]:
+        if param == variable:
```
Comment (jorshi) [inline transforms.py:99]: What's this case? If i'm following this correctly -- you select a random transform, and then select a random parameter with that transform -- and hold that parameter constant (leave at the default value)? Is that correct? What's the reason for that?

File: transforms.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        param, v = choose_value(param, low, hi)
+        params[param] = v
+    for param, vals in transform_spec[2]:
+        params[param] = random.choice(vals)
+
+    variable_values = [
+        choose_value(variable, variable_spec[1], variable_spec[2])[1]
+        for i in range(N_TRANSFORMS)
+    ]
+    variable_values.sort()
```
Comment (jorshi) [inline transforms.py:110]: And now create N_TRANSFORMS different perturbations of the selected variable. Ahhh I see. You select one parameter to create a bunch of different perturbations of -- and hold all the other parameters constant at some value that has been randomly selected above. Cool.

File: transforms.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            ("balance", [None, "e", "B", "b"]),
+        ],
+    ),
+]
+
+
+def choose_value(name, low, hi):
+    v = random.uniform(low, hi)
+    if name == "midi":
+        # MIDI is one of the few that is transformed to another scale, frequency
```
Comment (jorshi) [inline transforms.py:86]: nice

File: transforms.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    for outf2 in outfiles[1:]:
+        input = torch.tensor(sf.read(outf2)[0])
+        target = torch.tensor(sf.read(outfiles[0])[0])
+        if len(input) < len(target):
+            target = target[: len(input)]
+        else:
+            input = input[: len(target)]
+        dists.append(dist(input, target).item())
+    # print(dists)
+    print(scipy.stats.spearmanr(dists, range(len(dists)))[0])
```
Comment (jorshi) [inline transforms.py:141]: Whats going on here?

**Review by jorshi — APPROVED** [review]

**Review by jorshi — APPROVED** [review]

File: transforms.py (hunk 2 above)
Comment (jorshi) [reply transforms.py:110]: One thought - so we uniformly sample a single parameter for these perturbations. The goal being to have a range of different audio files with various levels of transform applied to it based on varying this parameter -- we want to have some small differences as well as large differences. What if we select a random value for the parameter, and then select the perturbations by drawing from a normal distribution with that value as the mean -- so we get varying degrees of difference from that center value. Then we can use the standard deviation to have more control over the degree of variation?

File: transforms.py (hunk 1 above)
Comment (turian) [reply transforms.py:99]: This is deprecated, see followup PR

File: transforms.py (hunk 2 above)
Comment (turian) [reply transforms.py:110]: Also deprecated. See overleaf and later PRs

File: transforms.py (hunk 4 above)
Comment (turian) [reply transforms.py:141]: Deprecated, turns out this ranking task is not hard to learn.
